### PR TITLE
[bugfix] Make sure the query rewriter does not silently fail 

### PR DESCRIPTION
**Pull request by ljo**
(or fail with maxClauseCount exceeded) but chooses the most suitable one depending on terms and docs quote which the static one does not do.

### CODE REVIEW

**Comment by adamretter** [comment]
@ljo Can you also provide a test case please?

**Comment by ljo** [comment]
Yes, I will add another commit to this and a test case.

**Comment by ljo** [comment]
JFYI I am also looking into a full migration to Lucene 5.

**Comment by adamretter** [comment]
@ljo thanks

**Comment by ljo** [comment]
@adamretter Now both implementation and tests, a junit one and a smoketest, are in place.

**Comment by adamretter** [comment]
@ljo Okay cool, I am reviewing

**Comment by adamretter** [comment]
@ljo Did you run the test suite? I am getting a huge amount of test failures.

**Comment by adamretter** [comment]
@ljo Okay I am being dumb, let me try again...

**Comment by ljo** [comment]
@adamretter Yes, I run the testsuite 5-10 times per day ...

**Comment by adamretter** [comment]
@ljo cool, all checks out here. Thanks.
